src/terms/terms.jl

```diff
@@ -80,42 +80,67 @@ end
 forces(scfres) = forces(scfres.basis, scfres.ψ, scfres.occupation, ρ=scfres.ρ)
 
 
-"""
-    compute_kernel(term::Term; kwargs...)
+@doc raw"""
+    compute_kernel(basis::PlaneWaveBasis; kwargs...)
 
 Computes a matrix representation of the full response kernel
 (derivative of potential with respect to density) in real space.
+For non-spin-polarized calculations the matrix dimension is
+`prod(basis.fft_size)` × `prod(basis.fft_size)` and
+for collinear spin-polarized cases it is
+`2prod(basis.fft_size)` × `2prod(basis.fft_size)`.
+In this case the matrix has effectively 4 blocks, which are:
+```math
+\left(\begin{array}{cc}
+    K_{α, \text{tot}} & K_{α, \text{spin}}\\
+    K_{β, \text{tot}} & K_{β, \text{spin}}
+\end{array}\right)
+```
+i.e. corresponding to a mapping
+``(ρ_\text{tot}, ρ_\text{spin}) = (ρ_α + ρ_β, ρ_α - ρ_β)^T ↦ (V_α, V_β)^T``.
 """
-function compute_kernel(term::Term; kwargs...)
-    false  # "strong zero": false + something = something
-end
-"""
-    apply_kernel(term::Term, dρ; kwargs...)
-
-Computes the potential response to a perturbation dρ in real space
-"""
-function apply_kernel(term::Term, dρ; kwargs...)
-    nothing  # by default, no kernel
-end
-
 function compute_kernel(basis::PlaneWaveBasis{T}; kwargs...) where {T}
-    k = zeros(T, prod(basis.fft_size), prod(basis.fft_size))
+    n_spin = basis.model.n_spin_components
+    kernel = zeros(T, n_spin * prod(basis.fft_size), n_spin * prod(basis.fft_size))
     for term in basis.terms
-        k .+= compute_kernel(term; kwargs...)
+        term == false && continue
```

Comment:
If we're not going to use the "strong zero" property, might as well return nothing there, no?